x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/create_integration_assistant.tsx

```diff
@@ -5,31 +5,96 @@
  * 2.0.
  */
 
-import React, { useReducer, useMemo, useEffect } from 'react';
+import React, { useReducer, useMemo, useEffect, useCallback } from 'react';
 import { KibanaPageTemplate } from '@kbn/shared-ux-page-kibana-template';
 import { Header } from './header';
 import { Footer } from './footer';
-import { ConnectorStep, isConnectorStepReady } from './steps/connector_step';
-import { IntegrationStep, isIntegrationStepReady } from './steps/integration_step';
-import { DataStreamStep, isDataStreamStepReady } from './steps/data_stream_step';
-import { ReviewStep, isReviewStepReady } from './steps/review_step';
-import { CelInputStep, isCelInputStepReady } from './steps/cel_input_step';
-import { ReviewCelStep, isCelReviewStepReady } from './steps/review_cel_step';
+import { useNavigate, Page } from '../../../common/hooks/use_navigate';
+import { ConnectorStep, isConnectorStepReadyToComplete } from './steps/connector_step';
+import { IntegrationStep, isIntegrationStepReadyToComplete } from './steps/integration_step';
+import { DataStreamStep, isDataStreamStepReadyToComplete } from './steps/data_stream_step';
+import { ReviewStep, isReviewStepReadyToComplete } from './steps/review_step';
+import { CelInputStep, isCelInputStepReadyToComplete } from './steps/cel_input_step';
+import { ReviewCelStep, isCelReviewStepReadyToComplete } from './steps/review_cel_step';
 import { DeployStep } from './steps/deploy_step';
 import { reducer, initialState, ActionsProvider, type Actions } from './state';
 import { useTelemetry } from '../telemetry';
 import { ExperimentalFeaturesService } from '../../../services';
 
+type StepName = '1' | '2' | '3' | '4' | 'cel_input' | 'cel_review' | 'deploy';
+
+const stepNames: Record<StepName, string> = {
+  '1': 'Connector Step',
+  '2': 'Integration Step',
+  '3': 'DataStream Step',
+  '4': 'Review Step',
+  cel_input: 'CEL Input Step',
```

Comment:
These are only temporary steps. I am working now on finishing up CEL generation being moved to the flyout during step 3 as specified by UX.

Reply:
Do we know how the telemetry will look like? Perhaps it will fit into the `CEL Input` / `CEL Review` scheme?

I can also just remove the new events – mostly I wanted to fix the state where we sent 'Deploy Step' for CEL step 5.

Reply:
It will mostly fit that pattern. If you look at the UX design (linked in the slack thread), you'll see the flow. It's essentially input spec -> select path and auth -> cel review. 

Up to you, I just didn't want you to waste your time fixing all this up, just for me to essentially refactor it all again :)

Reply:
It sounds like the refactor I did where we move things from the footer up to the main component was in any case necessary and we can also keep the new telemetry events. So I don't think any time was wasted.